percona-server/Dockerfile

```diff
@@ -1,51 +1,56 @@
 # also, we set debconf keys to make APT a little quieter
 RUN { \
-		echo percona-server-server-$PERCONA_MAJOR percona-server-server/root_password password 'unused'; \
-		echo percona-server-server-$PERCONA_MAJOR percona-server-server/root_password_again password 'unused'; \
-	} | debconf-set-selections \
-	&& apt-get update \
-	&& apt-get install -y --force-yes \
-		percona-server-server-$PERCONA_MAJOR=$PERCONA_VERSION \
-	&& rm -rf /var/lib/apt/lists/* \
+                echo percona-server-server-$PERCONA_MAJOR percona-server-server/root_password password 'unused'; \
+                echo percona-server-server-$PERCONA_MAJOR percona-server-server/root_password_again password 'unused'; \
+        } | debconf-set-selections \
+        && apt-get update \
+        && apt-get install --force-yes -y \
+                percona-server-server-$PERCONA_MAJOR=$PERCONA_VERSION percona-server-tokudb-$PERCONA_MAJOR=$PERCONA_VERSION\
+        && rm -rf /var/lib/apt/lists/* \
         && mv /etc/mysql/percona-server.conf.d/mysqld.cnf /etc/mysql/my.cnf \
 # comment out any "user" entires in the MySQL config ("docker-entrypoint.sh" or "--user" will handle user switching)
-	&& sed -ri 's/^user\s/#&/' /etc/mysql/my.cnf \
+        && sed -ri 's/^user\s/#&/' /etc/mysql/my.cnf \
 # purge and re-create /var/lib/mysql with appropriate ownership
-	&& rm -rf /var/lib/mysql && mkdir -p /var/lib/mysql /var/run/mysqld \
-	&& chown -R 1001:0 /var/lib/mysql /var/run/mysqld \
+        && rm -rf /var/lib/mysql && mkdir -p /var/lib/mysql /var/run/mysqld \
+        && chown -R 1001:0 /var/lib/mysql /var/run/mysqld \
 # ensure that /var/run/mysqld (used for socket and lock files) is writable regardless of the UID our mysqld instance ends up having at runtime
-	&& chmod 777 /var/run/mysqld
+        && chmod 777 /var/run/mysqld
 
 # comment out a few problematic configuration values
 # don't reverse lookup hostnames, they are usually another container
 RUN sed -Ei 's/^(bind-address|log)/#&/' /etc/mysql/my.cnf \
-	&& echo 'skip-host-cache\nskip-name-resolve' | awk '{ print } $1 == "[mysqld]" && c == 0 { c = 1; system("cat") }' /etc/mysql/my.cnf > /tmp/my.cnf \
-	&& mv /tmp/my.cnf /etc/mysql/my.cnf
+        && echo 'skip-host-cache\nskip-name-resolve' | awk '{ print } $1 == "[mysqld]" && c == 0 { c = 1; system("cat") }' /etc/mysql/my.cnf > /tmp/my.cnf \
+        && mv /tmp/my.cnf /etc/mysql/my.cnf
 
 VOLUME ["/var/lib/mysql", "/var/log/mysql"]
 
 RUN sed -Ei '/log-error/s/^/#/g' -i /etc/mysql/my.cnf
 
+RUN echo "\n[mysqld_safe]\nthp-setting=never" >> /etc/mysql/my.cnf
+RUN sed -i 's:mysqld_safe]:mysqld_safe]\nthp-setting=never:' /etc/mysql/my.cnf
+RUN sed -i "s:^  exit 1$::" $(which ps_tokudb_admin)
+RUN sed -i "s:JEMALLOC_STATUS=$?:JEMALLOC_STATUS=0:" $(which ps_tokudb_admin)
```

Comment:
In general ps_tokudb_admin must be executed from root. These hacks help is to execute it from mysql user.
Surely we can start docker container by default using root user but it is not recommended.
So it is easier to add such hacks. In this case we still use mysql user and have enabled tokudb engine.